src/roadmap.md

```diff
@@ -91,98 +51,129 @@ transactions and messages for Java Binding too. This step will promote all the
 corresponding [benefits](#Protobuf-as-the-Serialization-Format-in-Exonum)
 discussed above.
 
-## First Quarter of 2019
+### Java Binding: Time Service
 
-### Service Interface Standard: Read Requests
+We add another built-in Exonum service into the array of services that can be
+easily launched with an Exonum blockchain in Java – the Time Oracle. The Time
+Oracle allows user services to access the calendar time that validators supply
+to the blockchain.
 
-To increase the performance of the system, we will add read requests to the
-list of methods supported by interfaces of services. Read requests do not modify
-the blockchain state and can be handled locally on the full node which
-receives the request from a client.
+## Second Quarter of 2019
 
-### Pre-check of Transactions in the Memory Pool
+### Secure Workflow for Node Administration
 
-Checking the validity of transactions which are included into the memory pool
-is one of the methods for preventing DoS attacks. This check will also ensure
-that transactions do not lead to any errors when processed.
+The introduction of a workflow for administrative operations, such as the
+deployment of services, is aimed at improving the security of the system.
+The exact details of the workflow are being currently discussed. One of the
+promising ideas is devising a service through which all the administrative
+processes will be conducted.
 
-## Second Quarter of 2019
+### Merkle DB - Minimal Profile
 
-### Dynamic Java Services
+The first step within the global initiative to introduce Merkle DB in Exonum.
+This step will result in implementation of a more intuitive storage API
+interface. Such interface will simplify interaction with the storage for service
+developers.
 
-The introduction of dynamic Java services will enable adding Java services to
-a running blockchain without the need to restart nodes. In other words, new
-services will be included into the network on the go. Support for
-dependencies between dynamic services will be added in one of the following
-versions.
+### Java Application Package
 
-### Communication between Services within the Blockchain
+A usability achievement that presupposes shipment of Exonum as a
+ready-to-install application package. The package is targeted at Java developers
+who plan to develop Exonum-based services. The application will have Rust and
+other related libraries pre-installed. This means that no compilation of the
+application will be required.
 
-As a method of improving the modularity of services and expanding the
-possibilities for their reuse, the Exonum team will introduce the ability for
-services to communicate with one another within one blockchain. Services will
-be able to issue and process transactions from other services and can, thus, be
-reused in a variety of scenarios.
+### Java Light Client - v.1
```

Comment:
v.1 might seem confusing (as the described things will be released in 0.2.0, but some more features might come in 0.2+). Maybe name it just "Java light client", and the latter section "Java Light Client Enhancements"?